### PR TITLE
Fix widget in IEF forms

### PR DESCRIPTION
**Pull request by chrfritsch**

### CODE REVIEW

File: src/Plugin/Field/FieldWidget/EntityReferenceOverrideAutocompleteWidget.php [hunk 1]
```diff
@@ -201,19 +217,80 @@ public function formElement(FieldItemListInterface $items, $delta, array $elemen
...
+      '#ajax' => [
+        'callback' => [static::class, 'updateEntityReferenceOverrideWidget'],
+        'wrapper' => $wrapper_id,
+      ],
+      '#attributes' => [
+        'class' => ['js-hide'],
+        // This is used to pass the selection from the modal to the widget.
+        'data-entity-reference-override-update' => $field_widget_id,
+      ],
+      '#submit' => [[static::class, 'updateEntityReferenceOverrideFieldState']],
```
Comment (ol0lll) [inline src/Plugin/Field/FieldWidget/EntityReferenceOverrideAutocompleteWidget.php]: These function names are unnecessarily long, use 'updateWidget', 'updateFieldState'

File: src/Plugin/Field/FieldWidget/EntityReferenceOverrideAutocompleteWidget.php [hunk 2]
```diff
@@ -201,19 +217,80 @@ public function formElement(FieldItemListInterface $items, $delta, array $elemen
...
+      $values[$key]['overwritten_property_map'] = Json::decode($value['overwritten_property_map'] ?? '{}');
+    }
+
+    unset($values['add_more']);
+
+    $field_state = static::getWidgetState($element['#field_parents'], $element['#field_name'], $form_state);
+    $field_state['items'] = $values;
+    static::setWidgetState($element['#field_parents'], $element['#field_name'], $form_state, $field_state);
+
+    $form_state->setRebuild();
```
Comment (ol0lll) [inline src/Plugin/Field/FieldWidget/EntityReferenceOverrideAutocompleteWidget.php]: why do we rebuild dat?

**Review by ol0lll — COMMENTED** [review]
You should add some comment that this mechanism is copied form media library for the most part.

File: src/Plugin/Field/FieldWidget/EntityReferenceOverrideAutocompleteWidget.php (hunk 1 above)
Comment (chrfritsch) [reply src/Plugin/Field/FieldWidget/EntityReferenceOverrideAutocompleteWidget.php]: Media library has a updateWidget method. So when we combine the widget later, we will have a naming clash. That's the reason I made the name so unique.

File: src/Plugin/Field/FieldWidget/EntityReferenceOverrideAutocompleteWidget.php (hunk 2 above)
Comment (chrfritsch) [reply src/Plugin/Field/FieldWidget/EntityReferenceOverrideAutocompleteWidget.php]: Seems to work without as well. I will remove it

**Review by ol0lll — APPROVED** [review]

**Pull request merged**